types/rifat__miio/index.d.ts

```diff
@@ -0,0 +1,53 @@
+// Type definitions for @rifat/miio 2.0
+// Project: https://github.com/torifat/miio
+// Definitions by: Tadeusz Wyrzykowski <https://github.com/Shaquu>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+export interface Device {
+    on(eventName: string | symbol, listener: (...args: any[]) => void): this;
+}
+
+export interface AirPurifier3 extends Device {
```

Comment:
Add overloaded .matches method here:

matches(type: "type:air-purifier"): true
matches(type: string): boolean

Reply:
Will be added